src/expr/src/vector_op/arithmetic_op.rs

```diff
@@ -201,6 +216,31 @@ where
     interval_int_mul::<T2, T1, T3>(r, l)
 }
 
+#[inline(always)]
+pub fn date_time_add<T1, T2, T3>(
+    l: NaiveDateWrapper,
+    r: NaiveTimeWrapper,
+) -> Result<NaiveDateTimeWrapper> {
+    let date_time = NaiveDateTime::new(l.0, r.0);
+    let date_time_wrapper = NaiveDateTimeWrapper::new(date_time);
+    Ok(date_time_wrapper)
+}
+
+#[inline(always)]
+pub fn time_date_add<T1, T2, T3>(
+    l: NaiveTimeWrapper,
+    r: NaiveDateWrapper,
+) -> Result<NaiveDateTimeWrapper> {
+    date_time_add::<T2, T1, T3>(r, l)
+}
+
+#[inline(always)]
+pub fn time_time_sub<T1, T2, T3>(l: NaiveTimeWrapper, r: NaiveTimeWrapper) -> Result<IntervalUnit> {
+    let tmp = l.0 - r.0;
+    let ms = tmp.sub(Duration::days(tmp.num_days())).num_milliseconds();
```

Comment:
Why do we need this line? Do we have any edge case tesvting it? When will the diff be larger than 1 day? `24:00:00 - 0:00:00`?

```
postgres=# select time '24:00:00' - time '0:00:00';
 ?column?
----------
 24:00:00
(1 row)
```
This is the behavior of postgres.

Reply:
Yes it seems `NaiveTime` cannot store `24:00:00`... But I guess the original question is asking why not simply `let ms = tmp.num_milliseconds();`

Reply:
The cause of this error could be something like this， the parser we used is not support '24:00:00'
https://github.com/singularity-data/risingwave/blob/7808cacc85ebae4347469ac9a19c7700b1c0c570/src/expr/src/vector_op/cast.rs#L48

`parse_from_str` will return err  and the `NaiveTime` we inport in our project is not support `24:00:00`;

<img width="1028" alt="image" src="https://user-images.githubusercontent.com/65114384/175882526-2126bd33-8e36-41db-b9ce-b01aad250442.png">

Reply:
> Yes it seems `NaiveTime` cannot store `24:00:00`... But I guess the original question is asking why not simply `let ms = tmp.num_milliseconds();`

I think I forgot to change it when I copied the demo 🥹